### ISSUE TITLE
Transactions are not retried

### ISSUE THREAD

**Issue by dtarres** [sqlbunny/sqlbunny#19]
Go version: 1.16.4 in local and 1.15 in prod
DB: PostgreSQL 9.6

We are having issues with Atomic retries. We suspected that in our environment, where we use SQLBunny as our ORM, when transactions failed they were never retried, even though they should have.

After our investigation, we concluded that the function `shouldRetryTransaction` always returns `false`, due to error wrapping. The error returned is wrapped more than once, so SQLBunny doesn't detect that the transaction failed due to serialization failure or deadlock detected.
The following screenshot is obtained when debugging, inside `shouldRetryTransaction` (its code is copied below for easy access) and paused just after `err2` takes value when unwrapping the main error `err`.
![image](https://user-images.githubusercontent.com/36658191/119905362-68b45300-bf4c-11eb-9f29-7a9abb31f944.png)

```
func shouldRetryTransaction(err error) bool {
	err2 := errors.Unwrap(err)
	if err2, ok := err2.(*pq.Error); ok {
		n := err2.Code.Name()
		if n == "serialization_failure" || n == "deadlock_detected" {
			return true
		}
	}
	return false
}
```

In order to reproduce the error, the following code was executed. It basically creates two concurrent transactions that edit the same instance of User, and fail.
```
package atomictest

import (
	"context"
	"fmt"
	"sync"
	"testing"
	"time"

	"github.com/sqlbunny/sqlbunny/runtime/bunny"
	"github.com/stretchr/testify/assert"
)


func TestAtomic(t *testing.T) {
	t.Parallel()
	ctx := testhelper.Config(t)
	user := &models.User{
		ID: models.NewUserID(),
		FirstName: "",
	}
	require.NoError(t, user.Insert(ctx))
	nTriesAtomic := 0
	wg := sync.WaitGroup{}
	wg.Add(1)
	go func() {
		time.Sleep(1 * time.Second)
		_ = bunny.Atomic(ctx, func(ctx context.Context) error {
			user.FirstName = "lala2"
			_ = user.Update(ctx)
			time.Sleep(6 * time.Second)
			return nil
		})
		wg.Done()
	}()

	err := bunny.Atomic(ctx, func(ctx context.Context) error {
		nTriesAtomic++
		_, _ = models.FindUser(ctx, user.ID)
		time.Sleep(2 * time.Second)
		fmt.Println(nTriesAtomic)
		user.FirstName = "lala"
		return user.Update(ctx)
	})
	wg.Wait()
	assert.Error(t, err)
	assert.Equal(t, 12, nTriesAtomic) // always 1
}
```

**Comment by Dirbaio** [comment]
Thanks @dtarres for the report and the repro! Indeed, the error was not unwrapped enough.

Fixed in  fcf1e2f

**Comment by dtarres** [comment]
Many thanks for the quick response @Dirbaio! Looks nice, surely it fixes our problem. I'll get back to you tomorrow confirming it ^^
